### PR TITLE
Promote service load balancer finalizer to GA

### PR DESCRIPTION
**Pull request by MrHohn**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Ref kubernetes/enhancements#980 and kubernetes/cloud-provider#16. This PR promotes service load balancer finalizer to GA. There is no behavioral change.

**Which issue(s) this PR fixes**:

Fixes #NONE 

**Special notes for your reviewer**:
/assign @andrewsykim @bowei @thockin 

**Does this PR introduce a user-facing change?**:

```release-note
Finalizer Protection for Service LoadBalancers is now in GA (enabled by default). This feature ensures the Service resource is not fully deleted until the correlating load balancer resources are deleted.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/controller/service/controller_test.go [hunk 1]
```diff
@@ -307,8 +307,6 @@ func TestSyncLoadBalancerIfNeeded(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.desc, func(t *testing.T) {
-			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, serviceLoadBalancerFinalizerFeature, tc.enableFeatureGate)()
```
Comment (andrewsykim) [inline pkg/controller/service/controller_test.go]: Can we remove `serviceLoadBalancerFinalizerFeature` or add a TODO to remove in a follow-up PR?

Comment (MrHohn) [reply pkg/controller/service/controller_test.go]: Thanks Andrew. Totally, removed `serviceLoadBalancerFinalizerFeature` as that is no longer needed.

File: pkg/controller/service/controller.go [hunk 2]
```diff
@@ -348,15 +336,13 @@ func (s *Controller) syncLoadBalancerIfNeeded(service *v1.Service, key string) (
...
-		if utilfeature.DefaultFeatureGate.Enabled(serviceLoadBalancerFinalizerFeature) {
-			// Always try to add finalizer prior to load balancer creation.
-			// It will be a no-op if finalizer already exists.
-			// Note this also retrospectively puts on finalizer if the cluster
-			// is upgraded from a version that doesn't attach finalizer to a
-			// version that does.
-			if err := s.addFinalizer(service); err != nil {
-				return op, fmt.Errorf("failed to add load balancer cleanup finalizer: %v", err)
-			}
+		// Always try to add finalizer prior to load balancer creation.
```
Comment (andrewsykim) [inline pkg/controller/service/controller.go]: Wondering if this comment should be updated now. Maybe just state the intent now that it's a default behavior. 

```
// Always add a finalizer prior to creating load balancers, this ensures Services can't be deleted
// until all corresponding load balancer resources are also deleted.
```

Comment (andrewsykim) [reply pkg/controller/service/controller.go]: same for removeFinalizer above

**Review by andrewsykim — COMMENTED** [review]
minor comment, lgtm otherwise

File: pkg/controller/service/controller.go (hunk 2 above)
Comment (MrHohn) [reply pkg/controller/service/controller.go]: Yes that make more sense. Done.

File: pkg/controller/service/controller.go [hunk 3]
```diff
@@ -103,12 +98,13 @@ type serviceCache struct {
...
-	kubeClient          clientset.Interface
-	clusterName         string
-	balancer            cloudprovider.LoadBalancer
+	cloud            cloudprovider.Interface
+	knownHosts       []*v1.Node
+	servicesToUpdate []*v1.Service
+	kubeClient       clientset.Interface
+	clusterName      string
+	balancer         cloudprovider.LoadBalancer
+	// TODO(#85155): Stop relying on this and remove the cache completely.
```
Comment (MrHohn) [inline pkg/controller/service/controller.go]: BTW created https://github.com/kubernetes/kubernetes/issues/85155 for cleaning up the cache layer eventually.

**Comment by MrHohn** [comment]
/assign @thockin 
for approval on the feature gate. Thanks!

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**